python/tempo/tsdf.py

```diff
@@ -517,6 +519,32 @@ def resample(self, freq, func=None, metricCols = None, prefix=None, fill = None)
     enriched_tsdf = rs.aggregate(self, freq, func, metricCols, prefix, fill)
     return(enriched_tsdf)
 
+  def interpolate(self, target_cols: List[str], sample_freq: str, sample_func: str, fill: str, ts_col: str = None, partition_cols: List[str]=None):
```

Comment:
@guanjieshen , let's discuss why a frequency of function is required here at all. Users should be able to interpolate without resampling at all - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.interpolate.html

Reply:
Some suggestions: 

Change sample_freq -> freq (to be consistent with resample) 
Change sample_func -> func (to be consistent with resample) 
Make freq/func and target_cols optional  (you can use the code block below to populate the target_cols in case the user forgets or really just wants everything interpolated) 

```
    if target_cols is None:
        prohibited_cols = self.partitionCols + [self.ts_col]
        summarizable_types = ['int', 'bigint', 'float', 'double']
        # filter columns to find summarizable columns
        target_cols = [datatype[0] for datatype in self.df.dtypes if
                           ((datatype[1] in summarizable_types) and
                            (datatype[0].lower() not in prohibited_cols))]
```

Reply:
Made the changes to the naming, and modified the default behaviour for `target_cols` based on the code snipped.

For `freq` and `func` wouldn't these need to be required? Unless we are thinking that a user needs to call `resample` first before `interpolate`?